src/commands/run.rs

```diff
@@ -340,3 +353,53 @@ fn maybe_add_named_capability_to_store(
 
     Ok(())
 }
+
+#[cfg(test)]
+mod unittest {
+    use crate::commands::run::{handle_run, RunArgs};
+    use rand::Rng;
+    use slight_runtime::IORedirects;
+    use std::fs::File;
+    use std::path::{Path, PathBuf};
+    use tempfile::tempdir;
+    use tokio::fs;
+
+    #[tokio::test]
+    async fn test_handle_run_with_io() -> anyhow::Result<()> {
+        let module = "./src/commands/test/io-test.wasm";
+        assert!(Path::new(module).exists());
+        let slightfile = "./src/commands/test/slightfile.toml";
+        assert!(Path::new(slightfile).exists());
+
+        let tmp_dir = tempdir()?;
+        let stdin_path = tmp_dir.path().join("stdin");
+        let stdout_path = tmp_dir.path().join("stdout");
+        let stderr_path = tmp_dir.path().join("stderr");
+
+        let canary: String = rand::thread_rng()
+            .gen_ascii_chars()
+            .take(7)
+            .map(char::from)
+            .collect();
+        fs::write(&stdin_path, &canary).await?;
+        let _ = File::create(&stdout_path)?;
+        let _ = File::create(&stderr_path)?;
+
+        let args = RunArgs {
```

Comment:
It would be great to see an integration test using `slight run` command and passes in io redirects paths

Reply:
In the PR description, I left this as out of scope. I wasn't sure if it would be useful for anyone or if it would just pollute the command line args.